### ISSUE TITLE
Implement regularization

### ISSUE THREAD

**Issue by PetrToman** [encog/encog-java-core#28]
Hello,
please consider implementing regularization, as it is essential to deal with the overfitting problem.

I recommend watching 12 min. video "Regularization and Bias/Variance" of lesson X. ADVICE FOR APPLYING MACHINE LEARNING at https://class.coursera.org/ml/lecture/preview (Stanford ML course).

It would also be useful to enhance Encog Analyst - it could split data into 3 sets (training, cross validation, testing) and try to find the optimal regularization parameter automatically.

**Comment by joetanto** [comment]
I agree with @thomasj02, the bias terms must not be included when regularizing. I'd appreciate if someone can fix that. Thank you.

**Comment by vincenzodentamaro** [comment]
Well, allowing large biases gives our networks more flexibility in behaviour: large biases make it easier for neurons to saturate, which is sometimes desirable. So regularization on biases is not necessary.

**Comment by jeffheaton** [comment]
Since this was submitted, Encog has added dropout, L1 and L2.
